java/ql/automodel/src/README.md

```diff
@@ -0,0 +1,183 @@
+
+### Backwards Compatibility
+
+We try to keep changes to extraction queries backwards-compatible whenever feasible. There's several reasons:
+
+ - That automodel can always decide which version of the package to run is a flawed assumption. We don't have direct control over the version of the extraction queries running on the user's local machine.
+ - An automodel deployment will sometimes require the extraction queries to be published. If the new version of the extraction queries works with the old version of automodel, then it is much easier to roll back deployments of automodel.
+
+## Candidate Examples
+
+This section contains a few examples of the kinds of candidates that our queries might select, and why.
+
+:warning: For clarity, this section presents "candidates" that are **actual** sinks. Therefore, the candidates presented here would actually be selected as positive examples in practice - rather than as candidates.
+
+### Framework Mode Candidates
+
+Framework mode is special because in framework mode, we extract candidates (as well as examples) from the implementation of a framework or library while the resulting models are applied in code bases that are _using_ the framework or library.
+
+In framework mode, endpoints currently can have a number of shapes (see: `newtype TFrameworkModeEndpoint` in [AutomodelApplicationModeExtractCandidates.ql](https://github.com/github/codeql/blob/main/java/ql/automodel/src/AutomodelFrameworkModeCharacteristics.qll)). Depending on what kind of endpoint it is, the candidate is a candidate for one or several extensible types (eg., `sinkModel`, `sourceModel`).
+
+#### Framework Mode Sink Candidates
+
+Sink candidates in framework mode are inputs to calls. As, in framework mode, we work on the implementation of a callable, these inputs are represented by a method's parameter definition.
+
+For example, customer code could call the `Files.copy` method:
+
+```java
+// customer code using a library
+...
+Files.copy(userInputPath, outStream);
+...
+```
+
+In order for `userInputPath` to be modeled as a sink, the corresponding parameter must be selected as a candidate. In the following example, assuming they're not modeled yet, the parameters `source` and `out` would be candidates:
+
+```java
+// Files.java
+// library code that's analyzed in framework mode
+public class Files {
+  public static void copy(Path source, OutputStream out) throws IOException {
+    // ...
+  }
+}
+```
+
+#### Framework Mode Source Candidates
+
+Source candidates are a bit more varied than sink candidates:
+
+##### Parameters as Source Candidates
+
+A parameter could be a source, eg. when a framework passes user-controlled data to a handler defined in customer code.
```

Comment:
Good explanation. I see how we want parameters as both source and sink candidates, though in practice the probability of the _same_ parameter being both is I think slim.

Reply:
Minor

```suggestion
A parameter could be a source, e.g. when a framework passes user-controlled data to a handler defined in customer code.
```